gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -2325,3 +2261,89 @@ def test_initialize_client_w_{{transport_name}}():
 {% endfor %}{# method in service.methods.values() #}
 {% endif %}
 {% endmacro %}{# routing_parameter_test #}
+
+{# inteceptor_class_test generates tests for rest interceptors. #}
+{% macro inteceptor_class_test(service, method, transport, is_async) %}
+{% if not is_async %}{# TODO: Remove this guard once support for async rest interceptors is added. #}
+{% set await_prefix = get_await_prefix(is_async) %}
+{% set async_prefix = get_async_prefix(is_async) %}
+{% set async_decorator = get_async_decorator(is_async) %}
+{% set transport_name = get_transport_name(transport, is_async) %}
+{% set method_name = method.name|snake_case %}
+{% set async_method_prefix = "Async" if is_async else "" %}
+{{async_decorator}}
+@pytest.mark.parametrize("null_interceptor", [True, False])
+{{async_prefix}}def test_{{ method_name }}_{{transport_name}}_interceptors(null_interceptor):
+{% if 'grpc' in transport %}
+    raise NotImplementedError("gRPC is currently not supported for this test case.")
```

Comment:
Please add a TODO comment for the outstanding task of adding support for gRPC

Reply:
We don't support interceptors for `gRPC`. The guard is added to ensure that the macro isn't incorrectly called against the `gRPC` transport.